docs/best-practice.md

```diff
@@ -0,0 +1,255 @@
+	throw err;
+}
+```
+
+### Child
+
+Sometimes it makes sense to logically group log messages. We can achieve this with a [child logger](https://getpino.io/#/docs/child-loggers).
+Typically, you will need to add some identifying detail to associate the logs together, a good example would be using `uuid.v4()`.
+
+```javascript
+const log = logger.child({ someId: uuid.v4() });
+log.info('an informational message');
+```
+
+## Secrets
+
+> never store sensitive information in the repo. Always store it in an Azure Key Vault.
+> [Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/general/basic-concepts)
+
+The access rights to both of these Key Vaults are done on a least-privilege basis
+
+## ORM
+
+The ORM used by the application to access SQL Server is [Prisma](https://www.prisma.io/). The schema is defined in [schema.prisma](./packages/appeals-service-api/src/db/schema.prisma).
+
+**Note:** If the `prisma.schema` file has been updated, don't forget to run `npm run db:migrate:dev` to apply the changes.
+
+## Types
+
+Though we use JavaScript, which uses [dynamic & weak typing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#dynamic_and_weak_typing), we want to be type safe to help reduce bugs. To do this we define types using [jsdoc](https://jsdoc.app/), [Prisma](https://www.prisma.io/) (for database types), and [TypeScript](https://www.typescriptlang.org/) (only type defintion files, no code).
+
+### Prisma Types
+
+In the `packages/appeals-service/api` package, to import Prisma types in jsdoc, use
+
+```
+import('@prisma/client').MyType
+```
+
+Types for creates or updates may not have all fields, so use the `*CreateInput` or `*UpdateInput` variations for those:
+
+```
+import('@prisma/client').Prisma.MyTypeUncheckedCreateInput
+// or
+import('@prisma/client').Prisma.MyTypeUncheckedUpdateInput
+```
+
+If you need a type with a relation included, say an `AppealUser` with a `SecurityToken`, you can use:
+
+```
+import('@prisma/client').Prisma.AppealUserGetPayload<{include: {SecurityToken: true}}>
+```
+
+This type will have a `.SecurityToken` property with the fields from that model.
+
+## Data Models
+
+In Appeals FO we have multiple models or entities. Some comply with the PINS [data model](https://github.com/Planning-Inspectorate/data-model), others are for FO use only. There are a number of conventions it is helpful to be aware of:
+
+**Submissions**
+
+Submissions, such as appeal submissions, are authored in the Front Office by our users. We _submit_ them to the back office. This includes appeals, LPA questionnaires, and others. We own this data, and this data is only useful until it is submitted - once submitted we no longer show them to users, after that the back office version is the source of truth.
+
+> Note: currently appeal submissions are saved in Cosmos, and the collection is simply `appeals`. These will get migrated to SQL, and the model will likely be `AppealSubmission`
```

Comment:
has this migration happened?

Reply:
Not happening nice spot, should have read some of this more closely